compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -680,6 +680,12 @@ impl<'a> TraitDef<'a> {
                     param_clone
                 }
             })
+            .map(|mut param| {
+                // Remove all attributes, because there might be helper attributes
+                // from other macros that will not be valid in the expanded implementation.
+                param.attrs = ThinVec::new();
```

Comment:
The reason I wrote it like this, was that I assumed creating a new `ThinVec` doesn't allocate, and by creating a new one instead of just clearing the old one, we're telling the compiler that we don't need the allocated memory, which is a nano-optimization, if the compiler doesn't already optimize it away in either case haha.